### ISSUE TITLE
Make parse accept ReadableStream<Uint8Array>

### ISSUE THREAD

**Issue by king8fisher** [dbushell/xml-streamify#1]
`parse` seems to be the main working horse, it would be great to have the API to work with the `ReadableStream` as the source to allow for both File or any other types of streams.

**Comment by king8fisher** [comment]
Something along these lines:
```ts
export async function* parse(
  readable: ReadableStream<Uint8Array>,
  options?: ParseOptions,
): AsyncGenerator<Node, Node | void, void> {
  const document = new Node("@document");

  try {
    const init = { ...options?.fetchOptions };
    if (options?.signal) {
      init.signal = options.signal;
    }

    const stream = readable.pipeThrough(new XMLStream(), {
      signal: options?.signal,
    });
```

**Comment by dbushell** [comment]
This is a good idea thanks.

It is already possible to use a file URL like:
```js
parse('file:///path/to/file.xml');
```

but yes I agree also accepting a `ReadableStream` makes sense. I'll work on it this week.

**Comment by king8fisher** [comment]
Just tried that, it's quite hectic to form the path properly :D
But yes, I can confirm `file://....` works as if it's a web resource.

**Comment by dbushell** [comment]
Yeah paths can be a bit confusing! This gets passed to `fetch` which in Deno accepts `http:` `file:` and even `data:` URLs.

I've started a new branch and draft PR for this feature. I'll need to do more testing but the basic idea works.

**Comment by dbushell** [comment]
This is now implemented in v0.6.0. Apologies for taking so long!

**Comment by king8fisher** [comment]
Thank you! This is awesome
